lib/matplotlib/backends/_backend_tk.py

```diff
@@ -507,16 +507,21 @@ class NavigationToolbar2Tk(NavigationToolbar2, tk.Frame):
     Attributes
     ----------
     canvas : `FigureCanvas`
-        the figure canvas on which to operate
+        The figure canvas on which to operate.
     win : tk.Window
-        the tk.Window which owns this toolbar
-
+        The tk.Window which owns this toolbar.
+    pack_toolbar : bool, default: True
+        If True, add the toolbar to the parent's pack manager's packing list
+        during initialization with *side='bottom'* and *fill='x'*.
+        If you want to use the toolbar with a different layout manager, use
+        *pack_toolbar=False*.
     """
-    def __init__(self, canvas, window):
+    def __init__(self, canvas, window, pack_toolbar=True):
         self.canvas = canvas
         # Avoid using self.window (prefer self.canvas.get_tk_widget().master),
         # so that Tool implementations can reuse the methods.
         self.window = window
+        self.pack_toolbar = pack_toolbar
```

Comment:
This doesn't need to be a public attribute -- it's just there for coordination between `__init__` and `_init_toolbar` -- which gets called by `__init__` anyways.
In fact you could even move the call to `pack` out of `_init_toolbar` and put it here:
```
if pack_toolbar: self.pack(...)
```

Reply:
I moved the call.